solidity/ecdsa/contracts/WalletRegistry.sol

```diff
@@ -482,25 +484,37 @@ contract WalletRegistry is IRandomBeaconConsumer, Ownable {
         return dkg.hasDkgTimedOut();
     }
 
-    function getWallet(bytes32 publicKeyHash)
+    function getWallet(bytes32 walletID)
         external
         view
         returns (Wallets.Wallet memory)
     {
-        return wallets.registry[publicKeyHash];
+        return wallets.registry[walletID];
     }
 
-    /// @notice Checks if a wallet with the given public key hash is registered.
-    /// @param publicKeyHash Wallet's public key hash.
-    /// @return True if wallet is registered, false otherwise.
-    function isWalletRegistered(bytes32 publicKeyHash)
+    /// @notice Gets public key of a wallet with a given wallet ID.
+    ///         The public key is returned in an uncompressed format as a 64-byte
+    ///         concatenation of X and Y coordinates.
+    /// @param walletID ID of the wallet.
+    /// @return Uncompressed public key of the wallet.
+    function getWalletPublicKey(bytes32 walletID)
         external
         view
-        returns (bool)
+        returns (bytes memory)
     {
-        return wallets.isWalletRegistered(publicKeyHash);
+        return wallets.getWalletPublicKey(walletID);
     }
 
+    /// @notice Checks if a wallet with the given ID is registered.
+    /// @param walletID Wallet's ID.
+    /// @return True if wallet is registered, false otherwise.
+    function isWalletRegistered(bytes32 walletID) external view returns (bool) {
+        return wallets.isWalletRegistered(walletID);
+    }
+
+    // TODO: Add function to close the Wallet so the members are notified that
+    // they no longer need to track the wallet.
```

Comment:
In this PR or separately? If separately, please capture it in the issue and add it to [the project](https://github.com/orgs/keep-network/projects/31).

Reply:
https://github.com/keep-network/keep-core/issues/2864